flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/RexNodeExtractor.scala

```diff
@@ -455,7 +449,7 @@ class RexNodeToExpressionConverter(
 
       case TIMESTAMP_WITH_LOCAL_TIME_ZONE =>
         val v = literal.getValueAs(classOf[TimestampString])
-        toLocalDateTime(v).atZone(timeZone.toZoneId).toInstant
+        toLocalDateTime(v).atZone(ZoneId.of(ZoneOffset.UTC.getId)).toInstant
```

Comment:
Why we changed here？ For literal value of a TIMESTAMP_WITH_LOCAL_TIME_ZONE type field, the literal value represents the timestamp string under local time zone(i.e. session time zone), current behavior is ok for me.

Reply:
In the context of predicate pushdown，the literal has been converted to UTC before at the SQL to Rel phase. if do it again, it will lead to an incorrect result. Details in the link  https://issues.apache.org/jira/browse/FLINK-35318

Reply:
Thanks @lshangq for the information, evaluation once make sense to me.

Reply:
Thanks @leonardBang, could you please help merge this PR?